cloud/libvirt/actuators/machine/utils/domain.go

```diff
@@ -227,6 +231,73 @@ func setCoreOSIgnition(domainDef *libvirtxml.Domain, ignKey string) error {
 	return nil
 }
 
+const maxFWCFGChunkSize = 1024
+
+func breakUserDataIntoChunks(userData string, chunkMaxSize int) []string {
+	userDataLen := len(userData)
+	chunks := userDataLen / chunkMaxSize
+
+	b := make([]string, chunks)
+	for i := 0; i < chunks; i++ {
+		b[i] = userData[i*chunkMaxSize : (i+1)*chunkMaxSize]
+	}
+
+	if userDataLen%chunkMaxSize > 0 {
+		b = append(b, userData[chunks*chunkMaxSize:userDataLen])
+	}
+	return b
+}
+
+func setCloudInit(domainDef *libvirtxml.Domain, cloudInit *providerconfigv1.CloudInit, kubeClient kubernetes.Interface, machineNamespace string) error {
+	if cloudInit.ISO == "" {
```

Comment:
do we need to have this different from ign?

Reply:
IIRC, Ignition does not require `cloud-init` to be run. On the other hand, in order to run any user defined (at this point only bash) script, we need to run `cloud-init`. So ign handling is slightly different from the cloud init variant.